### PR TITLE
correct counting of resources in cluster and tenantinfo handlers

### PR DESCRIPTION
**Pull request by adameat**
### Changelog entry 

correct counting of resources in cluster and tenantinfo handlers

### Changelog category 

* Bugfix 
* Not for changelog (changelog entry is not required)

### CODE REVIEW

**Review by kruall — DISMISSED** [review]

File: ydb/core/base/pool_stats_collector.cpp [hunk 1]
```diff
@@ -51,14 +51,27 @@ class TStatsCollectingActor : public NActors::TStatsCollectingActor {
...
-            pools.emplace_back(pool.Name, pool.Usage, pool.Threads, pool.LimitThreads);
+            auto& pb = *systemUpdate->Record.AddPoolStats();
+            pb.SetName(pool.Name);
+            pb.SetUsage(pool.Usage);
+            pb.SetThreads(static_cast<ui32>(pool.Threads));
+            pb.SetLimit(static_cast<ui32>(pool.LimitThreads));
+            if (pool.Name != "IO") {
+                coresTotal += static_cast<ui32>(pool.DefaultThreads);
+            }
+            coresUsed += pool.Usage * pool.LimitThreads;
```
Comment (alexvru) [inline ydb/core/base/pool_stats_collector.cpp:69]: coresUsed can exceed coresTotal? :)

File: ydb/core/base/pool_stats_collector.cpp [hunk 2]
```diff
@@ -51,14 +51,27 @@ class TStatsCollectingActor : public NActors::TStatsCollectingActor {
     };
 
     void OnWakeup(const TActorContext &ctx) override {
+        static auto whiteboardId = NNodeWhiteboard::MakeNodeWhiteboardServiceId(ctx.SelfID.NodeId());
```
Comment (alexvru) [inline ydb/core/base/pool_stats_collector.cpp:54]: very VERY bad idea

File: ydb/core/viewer/json_handlers_viewer.cpp [hunk 3]
```diff
@@ -204,7 +204,7 @@ void InitViewerStorageUsageJsonHandler(TJsonHandlers &handlers) {
 }
 
 void InitViewerClusterJsonHandler(TJsonHandlers& handlers) {
-    handlers.AddHandler("/viewer/cluster", new TJsonHandler<TJsonCluster>(TJsonCluster::GetSwagger()), 4);
+    handlers.AddHandler("/viewer/cluster", new TJsonHandler<TJsonCluster>(TJsonCluster::GetSwagger()), 5);
```
Comment (alexvru) [inline ydb/core/viewer/json_handlers_viewer.cpp:207]: magic numbers :)

File: ydb/core/viewer/viewer_cluster.h [hunk 4]
```diff
@@ -342,8 +342,12 @@ class TJsonCluster : public TViewerPipeClient {
     }
 
     void InitSystemWhiteboardRequest(NKikimrWhiteboard::TEvSystemStateRequest* request) {
-        //request->AddFieldsRequired(-1);
-        Y_UNUSED(request);
+        for (auto field : {1, 2, 4, 5, 6, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 30, 31, 36, 37}) { // node_whiteboard.proto default fields
```
Comment (alexvru) [inline ydb/core/viewer/viewer_cluster.h:345]: maybe you want to use their enum names?

File: ydb/core/viewer/viewer_cluster.h [hunk 5]
```diff
@@ -462,23 +466,40 @@ class TJsonCluster : public TViewerPipeClient {
...
+        };
+
+        std::unordered_set<TString> hostPassed;
+        std::unordered_map<TString, TMemoryStats> memoryStats;
+
         for (TNode& node : NodeData) {
             const NKikimrWhiteboard::TSystemStateInfo& systemState = node.SystemState;
             (*ClusterInfo.MutableMapDataCenters())[node.DataCenter]++;
-            if (systemState.HasNumberOfCpus()) {
+            if (hostPassed.insert(systemState.GetHost()).second) {
```
Comment (alexvru) [inline ydb/core/viewer/viewer_cluster.h:480]: what about multiple nodes on the same host?

**Review by alexvru — CHANGES_REQUESTED** [review]

File: ydb/core/base/pool_stats_collector.cpp (hunk 2 above)
Comment (adameat) [reply ydb/core/base/pool_stats_collector.cpp:54]: I agree, didn't think about tests

File: ydb/core/base/pool_stats_collector.cpp (hunk 1 above)
Comment (adameat) [reply ydb/core/base/pool_stats_collector.cpp:69]: > coresUsed can exceed coresTotal? :)

technically - yes, they can. for example, when we have events taking longer than 1 second to process

File: ydb/core/viewer/json_handlers_viewer.cpp (hunk 3 above)
Comment (adameat) [reply ydb/core/viewer/json_handlers_viewer.cpp:207]: > magic numbers :)

versions :) for UI to know what to expect from the current backend

File: ydb/core/viewer/viewer_cluster.h (hunk 4 above)
Comment (adameat) [reply ydb/core/viewer/viewer_cluster.h:345]: > maybe you want to use their enum names?

maybe, maybe even form them based on protobuf reflection ... that's a good idea, by the way, thank you!

File: ydb/core/viewer/viewer_cluster.h (hunk 5 above)
Comment (adameat) [reply ydb/core/viewer/viewer_cluster.h:480]: > what about multiple nodes on the same host?

that's exactly what this fix made for!

**Review by alexvru — APPROVED** [review]

**Review by alexvru — APPROVED** [review]

**Pull request merged**